### PR TITLE
Update rustdoc.css

### LINKED ISSUE #120595: rustdoc: Long type names result in weird-looking documentation
Crates with long type names result in a strange-looking error, see [https://docs.rs/async-stripe/0.31.0/stripe/](https://docs.rs/async-stripe/0.31.0/stripe/)

image

I've tested this on Chrome and Firefox, and by building the docs locally from [https://github.com/arlyon/async-stripe](https://github.com/arlyon/async-stripe)

I believe the text is rendering vertically due to long type names like [https://docs.rs/async-stripe/0.31.0/stripe/enum.PaymentFlowsPrivatePaymentMethodsCardDetailsApiResourceEnterpriseFeaturesIncrementalAuthorizationIncrementalAuthorizationStatus.html](https://docs.rs/async-stripe/0.31.0/stripe/enum.PaymentFlowsPrivatePaymentMethodsCardDetailsApiResourceEnterpriseFeaturesIncrementalAuthorizationIncrementalAuthorizationStatus.html) which are auto-generated from OpenAPI.

Link to original issue [https://github.com/arlyon/async-stripe/issues/488](https://github.com/arlyon/async-stripe/issues/488)

### PR DESCRIPTION
**Pull request by BradMarr**
fixes #120595

removing this selector fixes weird wrapping in certain pages like https://docs.rs/async-stripe/0.31.0/stripe/index.html#structs

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
This indeed seems to fix the issue, thanks!

Please add a test in `tests/rustdoc-gui` to ensure this bug won't appear again later on. If you need help to write the test, don't hesitate to ask!

**Comment by BradMarr** [comment]
I'm not familiar with the test system and don't know I could prevent this issue from occurring in the future. The issue was simply that the property was spread to an element it shouldn't have been so I don't know if a test could detect a problem in the future without knowing the intention of the programmer.

**Comment by BradMarr** [comment]
I'm lost as to how an offset error would occur when my only change was to the overflow-wrap.

**Comment by BradMarr** [comment]
maybe we could do a minimum width along with the default method so that it only makes the lines overlap in that way if the edge case occurs

**Comment by BradMarr** [comment]
That would require parsing through all of the words, I think maybe giving each of the `.item-table>li>.item-name` a max width of say 600px it would allow for overflow of the itemname but only if it would otherwise cause a bug

**Comment by BradMarr** [comment]
How does that commit look?

File: src/librustdoc/html/static/css/rustdoc.css [hunk 1]
```diff
@@ -2059,6 +2060,10 @@ in src-script.js and main.js
 		background: var(--main-background-color);
 	}
 
+    .item-table>li>div {
+        max-width: unset !IMPORTANT;
```
Comment (GuillaumeGomez) [inline src/librustdoc/html/static/css/rustdoc.css:2064]: Why is this needed?

Comment (BradMarr) [reply src/librustdoc/html/static/css/rustdoc.css:2064]: On mobile, the name stretches the whole way and the description comes below it. If I maintain it using only 40vw it will crop only halfway throughout the screen.

File: src/librustdoc/html/static/css/rustdoc.css [hunk 2]
```diff
@@ -898,8 +898,8 @@ because of the `[-]` element which would overlap with it. */
 	So, for table layouts, override the default with break-word, which does
 	_not_ affect min-content intrinsic sizes.
 */
-table,
-.item-table {
+.item-table,
```
Comment (GuillaumeGomez) [inline src/librustdoc/html/static/css/rustdoc.css:901]: This change can be removed.

File: src/librustdoc/html/static/css/rustdoc.css (hunk 1 above)
Comment (GuillaumeGomez) [reply src/librustdoc/html/static/css/rustdoc.css:2064]: I see. Then in this case, please remove this rule and move the other into the media query for desktop (which is at [this line](https://github.com/rust-lang/rust/blob/master/src/librustdoc/html/static/css/rustdoc.css#L2117)).

**Comment by BradMarr** [comment]
I'm confused on how to fix the check issues.

**Comment by BradMarr** [comment]
> It says that in `tests/rustdoc-gui/type-declation-overflow.goml` at line 19, if expected a different value for the `offsetWidth` element property. Since you modified it, seems logical that it fails.

How should I go about fixing that?

**Comment by GuillaumeGomez** [comment]
Update the mentioned file.

**Comment by BradMarr** [comment]
> Update the mentioned file.

The mentioned file being the test or the css page?

**Comment by GuillaumeGomez** [comment]
The test.

File: tests/rustdoc-gui/type-declation-overflow.goml [hunk 3]
```diff
@@ -14,10 +14,6 @@ assert-property: ("body", {"scrollWidth": "1100"})
 // However, since there is overflow in the type declaration, its scroll width is bigger.
 assert-property: ("pre.item-decl", {"scrollWidth": "1324"})
 
-// In the table-ish view on the module index, the name should not be wrapped more than necessary.
```
Comment (GuillaumeGomez) [inline tests/rustdoc-gui/type-declation-overflow.goml:17]: Not remove, update.

**Comment by BradMarr** [comment]
How's this?

**Comment by neo-zip** [comment]
LGTM!

File: src/librustdoc/html/static/css/rustdoc.css [hunk 4]
```diff
@@ -2127,6 +2128,10 @@ in src-script.js and main.js
 		padding: 2px 4px;
 		box-shadow: 0 0 4px var(--main-background-color);
 	}
+
+	.item-table > li > .item-name {
+		max-width: 40vw;
```
Comment (GuillaumeGomez) [inline src/librustdoc/html/static/css/rustdoc.css:2133]: I just realized: why using the viewport width and not the parent element width?

**Comment by GuillaumeGomez** [comment]
Also once you're done, please [squash](https://blog.guillaume-gomez.fr/Git-tips/1/2) your commits.

File: src/librustdoc/html/static/css/rustdoc.css (hunk 4 above)
Comment (BradMarr) [reply src/librustdoc/html/static/css/rustdoc.css:2133]: The parents width element is not static so I can't make the css `50%`

Comment (GuillaumeGomez) [reply src/librustdoc/html/static/css/rustdoc.css:2133]: The viewport width isn't either? Not sure to understand what you mean.

Comment (BradMarr) [reply src/librustdoc/html/static/css/rustdoc.css:2133]: When the item-name gets a width of 50% nothing changes. I'm assuming this is due to the fact that the parent doesn't have a predefined width. Instead I've set it to 40vw which seems to scale pretty nicely.

Comment (GuillaumeGomez) [reply src/librustdoc/html/static/css/rustdoc.css:2133]: That doesn't make any sense. The parent has a width. Also: you can resize the sidebar, changing the parent element width, but not the viewport width.

Comment (BradMarr) [reply src/librustdoc/html/static/css/rustdoc.css:2133]: Upon interacting with the parents width, using something like `max-width: 50%` on `.item-name`, no change occurs. I'm not sure exactly why this is the case but I'm guessing it has to do with how the parents width is defined.

**Comment by BradMarr** [comment]
What if we did something like this, where instead of a table it is a flex. It does look a little different but I don't mind the appearance.

**Comment by GuillaumeGomez** [comment]
The failure is "normal", not related to your changes. I restarted failing checks. If it still fails with the same error, just rebase.

**Comment by GuillaumeGomez** [comment]
You messed up the git history. ^^'

**Comment by BradMarr** [comment]
I think I'm a little bit out of my depth here, somebody else should probably take the reins.

**Comment by GuillaumeGomez** [comment]
I'll take a look tomorrow.

**Comment by GuillaumeGomez** [comment]
Closing in favor of https://github.com/rust-lang/rust/pull/127418.
